src/v1alpha3/echo_client.ts

```diff
@@ -0,0 +1,233 @@
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// ** This file is automatically generated by gapic-generator-typescript. **
+// ** https://github.com/googleapis/gapic-generator-typescript **
+// ** All changes to this file may be overwritten. **
+
+import * as gax from 'google-gax';
+import * as path from 'path';
+import * as protobuf from 'protobufjs';
+
+import * as packageJson from '../../package.json';
+import * as protos from '../../pbjs-genfiles/protos';
+import * as gapicConfig from './echo_client_config.json';
+
+const version = packageJson.version;
+
+export interface ClientOptions extends gax.GrpcClientOptions,
+                                       gax.GoogleAuthOptions,
+                                       gax.ClientStubOptions {
+  libName?: string;
+  libVersion?: string;
+  clientConfig?: gax.ClientConfig;
+}
+
+interface Descriptors {
+  page: {[name: string]: gax.PageDescriptor};
+  stream: {[name: string]: gax.StreamDescriptor};
+  longrunning: {[name: string]: gax.LongrunningDescriptor};
+}
+
+export interface Callback<ResponseObject> {
+  (err: Error|null|undefined, value?: ResponseObject|null): void;
+}
+
+export interface PaginationCallback<
+    RequestObject, ResponseObject, ResponseType> {
+  (err: Error|null, values?: ResponseType[], nextPageRequest?: RequestObject,
+   rawResponse?: ResponseObject): void;
+}
+
+export interface PaginationResponse<
+    RequestObject, ResponseObject, ResponseType> {
+  values?: ResponseType[];
+  nextPageRequest?: RequestObject;
+  rawResponse?: ResponseObject;
+}
+
+export class EchoClient {
+  _descriptors: Descriptors = {page: {}, stream: {}, longrunning: {}};
+  _innerApiCalls: {[name: string]: Function};
+  auth: gax.GoogleAuth;
+
+  constructor(opts: ClientOptions) {
+    // Ensure that options include the service address and port.
+    opts = Object.assign(
+        {
+          clientConfig: {},
+          port: (this.constructor as typeof EchoClient).port,
+          servicePath: (this.constructor as typeof EchoClient).servicePath,
+        },
+        opts);
+
+    // Create a `gaxGrpc` object, with any grpc-specific options
+    // sent to the client.
+    opts.scopes = (this.constructor as typeof EchoClient).scopes;
+    const gaxGrpc = new gax.GrpcClient(opts);
+
+    // Save the auth object to the client, for use by other methods.
+    this.auth = gaxGrpc.auth;
+
+    // Determine the client header string.
+    const clientHeader = [
+      `gl-node/${process.version}`,
+      `grpc/${gaxGrpc.grpcVersion}`,
+      `gax/${gax.version}`,
+      `gapic/${version}`,
```

Comment:
Can we start to include the version of node.js being used with this data?

Reply:
Isn't it in the first header? `${process.version}` :)

Reply:
I am not a smart man